src/unix/core.c

```diff
@@ -1018,6 +1021,10 @@ int uv_os_homedir(char* buffer, size_t* size) {
   size_t len;
   long initsize;
   int r;
+#if defined(__ANDROID_API__) && __ANDROID_API__ < 21
+  typedef int(*fptr_getpwuid_r)(uid_t, struct passwd*, char*, size_t, struct passwd**);
```

Comment:
This line is too long, it should be <= 80 columns.  What you could do is:
```c
int (*getpwuid_r)(uid_t, struct passwd*, char*, size_t, struct passwd**);
// further down below
getpwuid_r = dlsym(RTLD_DEFAULT, "getpwuid_r");
```
That way, you don't need the third `#if`, you can just call `getpwuid_r(...)`.